src/plugins/data/common/search/search_source/create_search_source.ts

```diff
@@ -28,6 +28,7 @@
  * under the License.
  */
 
+import { QueryStringContract } from 'src/plugins/data/public/';
```

Comment:
unfortunately we should import public into common folder. the compiler might complain about this. 

we should consider taking the required function from the query string service (cache dataset) and exporting it in the common folder so that it's reusable by the dataset service and search source service

Reply:
Where would i see it complaining?

Reply:
I am surprised too that CI is not failing.